extensions/azurecore/src/utils.ts

```diff
@@ -161,6 +166,115 @@ export async function updateTenantIgnoreList(tenantIgnoreList: string[]): Promis
 	await configuration.update(constants.Filter, tenantIgnoreList, vscode.ConfigurationTarget.Global);
 }
 
+export function updateProviderSettings(defaultSettings: ProviderSettings[]): ProviderSettings[] {
+	let providerSettingsJson: ProviderSettingsJson[] | undefined = vscode.workspace.getConfiguration(constants.AzureSection).get(constants.ProviderSettingsJson) as ProviderSettingsJson[];
+	vscode.workspace.onDidChangeConfiguration(async (changeEvent) => {
+		const impactProvider = changeEvent.affectsConfiguration(constants.ProviderSettingsJsonSection);
+		if (impactProvider === true) {
+			await displayReloadAds(constants.ProviderSettingsJsonSection);
+		}
+	});
+	if (providerSettingsJson && providerSettingsJson.length > 0) {
+		try {
+			for (let cloudProvider of providerSettingsJson) {
+				// build provider setting
+				let newSettings = buildProviderSettings(cloudProvider);
+				defaultSettings.push(newSettings)
+				Logger.info(`Custom provider settings loaded for ${cloudProvider.settings.metadata.displayName}`);
+			}
+			void vscode.window.showInformationMessage(localize('providerSettings.success', 'Successfully loaded custom endpoints from settings'));
+
+		} catch (error) {
+			void vscode.window.showErrorMessage(localize('providerSettings.error', 'could not load endpoints from settings'));
```

Comment:
Make sure the error is logged to console - and consider either putting the error info in the error toast or at least direct them to check the logs for more details.

```suggestion
			void vscode.window.showErrorMessage(localize('providerSettings.error', 'Could not load endpoints from settings'));
```